### ISSUE TITLE
@angular/cdk dependency should be declared as peerDependency

### ISSUE THREAD

**Issue by dbusacca** [DanielYKPan/date-time-picker#375]
The `@angular/cdk` dependency should be declared as  peerDependency.

The `@angular/material` project declares it as peerDependency.

Otherwise if the main project requires a different versions of the cdk, npm will install a second copy of the cdk in the node_modules directory of the ng-pick-datetime.

Then an AOT build will fail at runtime with an `Attempting to attach an unknown Portal type. BasePortalOutlet accepts either a ComponentPortal or a TemplatePortal.` error because the ng-pick-datetime will be using a differente PortalComponent class than the application PortalModule.

**Comment by DanielYKPan** [comment]
What version of the cdk would cause this problem `Attempting to attach an unknown Portal type. BasePortalOutlet accepts either a ComponentPortal or a TemplatePortal.` ?

**Comment by dbusacca** [comment]
It's not a cdk version problem. The ng-pick-datetime module works great with every cdk/6.x.x version I've used.

In the project package.json I've declared

> "dependencies": {
>    `"@angular/cdk": "~6.2.0",`
>    `""ng-pick-datetime": "^6.0.3",`
> }

After `npm install`

In the directory
`node_module/@angular/cdk`
there will be the `@angular/cdk` version 6.2.0

In the directory
`node_module/ng-pick-datetime/node_modules/@angular/cdk`
there will be the `@angular/cdk` version 6.0.0

The problem is that with the AOT build the in the resulting bundle the ComponentPortal used by `ng-pick-datetime` will be the one from version 6.0.0.

The `@angular/cdk` PortalModule used by the RootModule will be version 6.2.0.

At runtime when the version 6.2.0 PortalModule will try to manage the datapicker portal it will check if `datapickerPortal instanceof ComponentPortal`

This check will fail because the PortalModule is using ComponentPortal version 6,2.0 but the datepicker is using ComponentPortal version 6.0.0.  In the application bundle there will be two different prototype code for the ComponentPortal class.

If `@angular/cdk` is declared as peerDependency there will be only and only one version in the application.
